core/editable.js

```diff
@@ -1215,10 +1215,11 @@
 
 						var backspace = key == 8,
 							range = sel.getRanges()[ 0 ],
-							startPath = range.startPath();
+							startPath = range.startPath(),
+							skipRangeTrimming = true;
 
 						if ( range.collapsed ) {
-							if ( !mergeBlocksCollapsedSelection( editor, range, backspace, startPath ) ) {
+							if ( !mergeBlocksCollapsedSelection( editor, range, backspace, startPath, skipRangeTrimming ) ) {
```

Comment:
This is the only call of the `mergeBlocksCollapsedSelection()`, so we can initialize `skipRangeTrimming` parameter inside it.